tests/testthat/setup_test_packages.R

```diff
@@ -18,34 +18,33 @@ dir.create(lib <- tempfile("ct_"), recursive = TRUE)
 
 tests <- normalizePath(testthat::test_path())
 pkg_dirs <- list(
-  examplepkg = system.file("examplepkg", package = "covtracer"),
-  no.evaluable.code = file.path(tests, "packages", "no.evaluable.code"),
-  reexport.srcref = file.path(tests, "packages", "reexport.srcref")
+  system.file("examplepkg", package = "covtracer"),
+  file.path(tests, "packages", "list.obj"),
+  file.path(tests, "packages", "no.evaluable.code"),
+  file.path(tests, "packages", "reexport.srcref")
 )
 
 # install our testing packages into a temp directory
+install.packages(
+  as.character(pkg_dirs),
+  type = "source",
+  lib = lib,
+  repos = NULL,
+  dependencies = FALSE,
+  INSTALL_opts = c("--with-keep.source", "--install-tests"),
+  quiet = TRUE
+)
```

Comment:
I had the bright idea of cleaning up our testing setup, but it sent me down a rabbit hole of rediscovering how a package with no source code crashes our test suite.